### PR TITLE
Implemented bug algorithms

### PR DESCRIPTION
**Pull request by sarimmehdi**
Implemented bug 0, bug 1 and bug 2 planning algorithms

### CODE REVIEW

**Comment by AtsushiSakai** [comment]
@sarimmehdi Hi. Thank you for your interesting PR!. Before core review, could you please add a test file for it? 
This test code might help you:
https://github.com/AtsushiSakai/PythonRobotics/blob/master/tests/test_a_star.py

**Comment by sarimmehdi** [comment]
@AtsushiSakai I have added the required file, please have a look. Thanks!

File: PathPlanning/Bug_Planning/bug.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+Bug Planning
+author: Sarim Mehdi(muhammadsarim.mehdi@studio.unibo.it)
+Source: https://sites.google.com/site/ece452bugalgorithms/
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+class Bug():
```
Comment (AtsushiSakai) [inline PathPlanning/Bug_Planning/bug.py:11]: I think `()` is unnecessary

File: PathPlanning/Bug_Planning/bug.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    def __init__(self, startx, starty, goalx, goaly, obsx, obsy):
```
Comment (AtsushiSakai) [inline PathPlanning/Bug_Planning/bug.py:12]: variable name convention: could you please all variable should use `_` between words. For example,
startx -> start_x
obsx -> obs_x
Could you please fix all variable names?

File: PathPlanning/Bug_Planning/bug.py [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+                        if cx == v_x and cy == v_y:
+                            use_pt = False
+                            break
+                    if use_pt:
+                        return cx, cy, False
+                if not use_pt:
+                    break
+        return self.rx[-1], self.ry[-1], True
+
+    def bug0(self):
```
Comment (AtsushiSakai) [inline PathPlanning/Bug_Planning/bug.py:53]: What is the difference between bug0 or bug1 or bug2? Could you please add docstring which explains the differences?

File: PathPlanning/Bug_Planning/bug.py (hunk 1 above)
Comment (AtsushiSakai) [inline PathPlanning/Bug_Planning/bug.py:11]: The class name should be `BugPlanner`?

File: PathPlanning/Bug_Planning/bug.py [hunk 4]
```diff
@@ -0,0 +1,296 @@
+"""
```
Comment (AtsushiSakai) [inline PathPlanning/Bug_Planning/bug.py:1]: Could you please change the directory `Bug_Planning` to `BugPlanning`?

File: PathPlanning/Bug_Planning/bug.py [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+                    if use_pt:
+                        return cx, cy, False
+                if not use_pt:
+                    break
+        return self.rx[-1], self.ry[-1], True
+
+    def bug0(self):
+        mov_dir = 'normal'
+        candx, candy = -9999, -9999
+        plt.plot(self.obsx, self.obsy, ".k")
```
Comment (AtsushiSakai) [inline PathPlanning/Bug_Planning/bug.py:56]: Could you please plotting function calls are covered by `show_animation` like:
https://github.com/AtsushiSakai/PythonRobotics/blob/fa227083372445f0f440b5d5ad878d9201ed7476/PathPlanning/AStar/a_star.py#L87-L93

**Review by AtsushiSakai — CHANGES_REQUESTED** [review]

**Comment by sarimmehdi** [comment]
@AtsushiSakai I do not understand the codecov/project error I am getting above. Can you please have a look and tell me what I need to do?

**Comment by AtsushiSakai** [comment]
You don’t need to care about codecov error. Next review is last one. I will merge it after you fixed it.

File: PathPlanning/BugPlanning/bug.py [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+                        return c_x, c_y, False
+                if not use_pt:
+                    break
+        return self.r_x[-1], self.r_y[-1], True
+
+    '''Bug0: Greedy algorithm where you move towards goal
+    until you hit an obstacle. Then you go around it
+    (pick an arbitrary direction), until it is possible
+    for you to start moving towards goal in a greedy manner again'''
+    def bug0(self):
```
Comment (AtsushiSakai) [inline PathPlanning/BugPlanning/bug.py:59]: Docstring usually is under function definition. Please check it 
https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard

File: PathPlanning/BugPlanning/bug.py [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+
+    '''Bug1: Move towards goal in a greedy manner.
+    When you hit an obstacle, you go around it and
+    back to where you hit the obstacle initially.
+    Then, you go to the point on the obstacle that is
+    closest to your goal and you start moving towards
+    goal in a greedy manner from that new point.'''
+    def bug1(self):
+        mov_dir = 'normal'
+        cand_x, cand_y = -9999, -9999
```
Comment (AtsushiSakai) [inline PathPlanning/BugPlanning/bug.py:122]: If 9999 is invalid number, you should use None or inf instead i think.

**Review by AtsushiSakai — CHANGES_REQUESTED** [review]

**Comment by sarimmehdi** [comment]
@AtsushiSakai I have implemented the requested changes. Please have a look.

**Comment by AtsushiSakai** [comment]
LGTM. Thank you!!

**Pull request merged**